server/app/views/admin/programs/ProgramPredicateConfigureView.java

```diff
@@ -524,8 +525,23 @@ private DivTag createScalarDropdown(
       }
     }
 
+    if (questionDefinition.isName()) {
+      try {
+        scalars = Scalar.getScalars(questionDefinition.getQuestionType());
+        // Filter out NAME_SUFFIX
+        scalars =
+            scalars.stream()
+                .filter(scalar -> !scalar.equals(Scalar.NAME_SUFFIX))
+                .collect(toImmutableSet());
+      } catch (InvalidQuestionTypeException | UnsupportedQuestionTypeException e) {
+        return div()
+            .withText("Sorry, you cannot create a show/hide predicate with this question type.");
+      }
+    }
+
     ImmutableList<OptionTag> options =
         scalars.stream()
+            .filter(scalar -> !questionDefinition.isName() || !scalar.equals(Scalar.NAME_SUFFIX))
```

Comment:
**suggestion:** When we make the proposed change, we wouldn't need this anymore.